server/pull_request.go

```diff
@@ -433,26 +436,28 @@ func (s *Server) CleanOutdatedPRs() {
 	}
 	prs := result.Data.([]*model.PullRequest)
 
-	mlog.Info("Will process the PRs", mlog.Int("PRs Count", len(prs)))
+	mlog.Info("Processing PRs", mlog.Int("PRs Count", len(prs)))
 
+	ctx, cancel := context.WithTimeout(context.Background(), requestTimeoutGithub)
+	defer cancel()
 	for _, pr := range prs {
-		pull, _, errPull := s.GithubClient.PullRequests.Get(context.Background(), pr.RepoOwner, pr.RepoName, pr.Number)
-		if errPull != nil {
-			mlog.Error("Error getting Pull Request", mlog.String("RepoOwner", pr.RepoOwner), mlog.String("RepoName", pr.RepoName), mlog.Int("PRNumber", pr.Number), mlog.Err(errPull))
-			if _, ok := errPull.(*github.RateLimitError); ok {
-				mlog.Error("GitHub rate limit reached")
-				s.CheckLimitRateAndSleep()
-			}
+		pull, _, err := s.GithubClient.PullRequests.Get(ctx, pr.RepoOwner, pr.RepoName, pr.Number)
+		if _, ok := err.(*github.RateLimitError); ok {
+			s.sleepUntilRateLimitAboveTokenReserve()
```

Comment:
This will sleep for a most an hour. At this point `ctx` got canceled.

With all the CLA changes I'm wondering if these changes are still needed. Maybe it's better to stash them and come back if needed.

Reply:
This is the cron cleaning up the database at night with outdated PRs. It will just return now, when it runs into the rate limit and not continue cleaning up.